### PR TITLE
MM-27315 Cypress tests for Integrations > Slash command auto-complete (7 test cases)

### PR DESCRIPTION
**Pull request by jfrerich**
#### Summary
Cypress tests for MM-27315 

Test Cases: 
[MM-T2829](https://mattermost.atlassian.net/projects/MM?selectedItem=com.atlassian.plugins.atlassian-connect-plugin%3Acom.kanoah.test-manager__main-project-page#!/testCase/MM-T2829)
[MM-T2830](https://mattermost.atlassian.net/projects/MM?selectedItem=com.atlassian.plugins.atlassian-connect-plugin%3Acom.kanoah.test-manager__main-project-page#!/testCase/MM-T2830)
[MM-T2831](https://mattermost.atlassian.net/projects/MM?selectedItem=com.atlassian.plugins.atlassian-connect-plugin%3Acom.kanoah.test-manager__main-project-page#!/testCase/MM-T2831)
[MM-T2832](https://mattermost.atlassian.net/projects/MM?selectedItem=com.atlassian.plugins.atlassian-connect-plugin%3Acom.kanoah.test-manager__main-project-page#!/testCase/MM-T2832)
[MM-T2833](https://mattermost.atlassian.net/projects/MM?selectedItem=com.atlassian.plugins.atlassian-connect-plugin%3Acom.kanoah.test-manager__main-project-page#!/testCase/MM-T2833)
[MM-T2834](https://mattermost.atlassian.net/projects/MM?selectedItem=com.atlassian.plugins.atlassian-connect-plugin%3Acom.kanoah.test-manager__main-project-page#!/testCase/MM-T2834)
[MM-T2835](https://mattermost.atlassian.net/projects/MM?selectedItem=com.atlassian.plugins.atlassian-connect-plugin%3Acom.kanoah.test-manager__main-project-page#!/testCase/MM-T2835)

#### Additional Notes
I created the `integrations/slash_commands` folder for these new tests

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-27315

### CODE REVIEW

**Comment by saturninoabril** [comment]
Thanks @jfrerich! Sorry for the delay. Will find time to review later today (just finishing somethig).

**Comment by jfrerich** [comment]
No worries @saturninoabril.  I can't imagination how busy y'all are with hacktoberfest

**Comment by DHaussermann** [comment]
- Tested to ensure all tests work
- Ensured what the tests are doing matches the TM4J intentions - For simplicity different pluign was used on dynamic argument that is more practical than the TM4J test

This test relies on code in the demo plugin that has not yet been released. Once the next release is published, this test should be changed to install the plugin via URL. Until then these tests cannot return green without manual intervention.
@saturninoabril I have added a do no merge label for now as these tests would not be able to be promoted to production until we have a Demo plugin release Jason can use.  Let me know if you think tis is okay or have alternative suggestions.

**Comment by saturninoabril** [comment]
Thanks @DHaussermann! I agree. I don't have any other suggestion than to have the new demo plugin release soon.

**Comment by DHaussermann** [comment]
@jfrerich we now have a Demo Plugin release `0.9.0` that you can install via the release URL.
Is it possible for you to implement this change?

**Comment by jfrerich** [comment]
Hey @DHaussermann! I have implemented installing the plugins via the URL.  This is ready for re-review!

File: e2e/cypress/integration/slash_commands/autocomplete_spec.js [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+            },
+        });
+    });
+
+    after(() => {
+        cy.apiRemovePluginById(pluginIdDemo);
+        cy.apiRemovePluginById(pluginIdJira);
+    });
+
+    it('T2829 Test an example of plugin that uses sub commands', () => {
```
Comment (saturninoabril) [inline e2e/cypress/integration/slash_commands/autocomplete_spec.js:50]: Please include full TM4J ID. Same for other `it` blocks.
```suggestion
    it('MM-T2829 Test an example of plugin that uses sub commands', () => {
```

File: e2e/cypress/integration/slash_commands/autocomplete_spec.js [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+        // # Post a slash command with trailing space
+        cy.get('#post_textbox').clear().type('/jira ');
+
+        // * Verify suggestion includes instance subcommand and is visible
+        cy.get('#suggestionList').should('contain.text', 'instance').scrollIntoView().should('be.visible');
+
+        // # Narrow down list to show info only suggestion
+        cy.get('#post_textbox').type('i');
+
+        // * Verify suggestion list is visible with at three children (issue, instance, info)
```
Comment (saturninoabril) [inline e2e/cypress/integration/slash_commands/autocomplete_spec.js:88]:
```suggestion
        // * Verify suggestion list is visible with three children (issue, instance, info)
```

**Review by saturninoabril — CHANGES_REQUESTED** [review]
Thanks @jfrerich for updating! Tested and passed. Left couple comments and I'm all good after that.

**Review by saturninoabril — APPROVED** [review]
Thanks Jason! LGTM.

**Review by DHaussermann — APPROVED** [review]
Tested last round of changes.
LGTM!

Moved some of the corresponding TM4J tickets form `ticket open` to `update`

**Pull request merged**